crates/matrix-sdk/src/widget_api/handler/driver.rs

```
@@ -36,8 +34,8 @@ impl<'t> From<&'t OpenIDState> for openid::State {
 impl From<OpenIDResult> for openid::State {
     fn from(result: OpenIDResult) -> Self {
         match result {
-            Some(response) => openid::State::Allowed(response),
-            None => openid::State::Blocked,
+            Ok(response) => openid::State::Allowed(response),
+            Err(_) => openid::State::Blocked,
```

Ignoring an error here suggests that we do want `Option` and not `Result` 🙂 Otherwise we would use the error. Because what's the purpose of having a `Result` on a library level if this error never (1) gets used inside a library and (2) never gets to the user/application, i.e. cannot be used.

---

there are different errror that can occur however. here: https://github.com/toger5/matrix-rust-sdk/blob/8dfbe061353a67be6a00d4dec9865077bc6eb323/crates/matrix-sdk/src/widget_api/driver/mod.rs#L66-L89

Maybe we want blocked to contain an Option<Error>?

---

I updated it to at least print the error and the context of the error so we know why the oidc token state is blocked. Did we fail to get the token because of an error or because the user did not grant permission.

---

Explained in the 1:1 🙂 Can take it over. Short summary: seems like there was no need to use `Error` there (the original implementation simply used `Option`). Printing the error is also not useful, because it's a library code, not the application code.